pygmt/datasets/map_tiles.py

```diff
@@ -0,0 +1,109 @@
+"""
+Function to load raster basemap tiles from XYZ tile providers, and load as
+:class:`xarray.DataArray`.
+"""
+
+try:
+    import contextily
+except ImportError:
+    contextily = None
+
+import numpy as np
+import xarray as xr
+
+__doctest_requires__ = {("load_map_tiles"): ["contextily"]}
```

Comment:
Found a nice way to skip the doctest when `contextily` is not installed (e.g. in the `Python 3.8 / NumPy 1.21` tests) using https://github.com/astropy/pytest-doctestplus/tree/v0.12.1#doctest-dependencies. Thanks to #1790 for adding `pytest-doctestplus`!